### DISCUSSION TITLE
Bevy change bnackend

### DISCUSSION THREAD

**Discussion by ilumar589** [bevyengine/bevy#3875] [Q&A] [answered]
I've asked this on discord already so I'll just copy paste. Hi. I have a reported problem with the vulkan backend https://github.com/bevyengine/bevy/issues/3780#issuecomment-1028574151 . Is there any way to force bevy and wgpu to use a different backend ? Or force them to enable the semaphore feature mentioned in the error trace ? I really want to continue my bevy journey and this is a blocker for me 🙂

**Comment by ickk** [comment, marked as answer]
Yes! You ought to be able to insert a `WgpuOptions` resource that selects a specific wgpu backend. 

Example demonstrating how to insert the resource: https://github.com/bevyengine/bevy/blob/main/examples/app/headless_defaults.rs

The list of wgpu backends: https://wgpu.rs/doc/wgpu/struct.Backends.html

**Comment by heavyrain266** [comment]
You can set it this way with simple variable and resource. 

Source : [here](https://github.com/HeavyRain266/space_invaders/blob/master/src/main.rs)

 ```rs
 ​        ​// Setup native Graphics API for each platform. 
 ​        ​let​ platform_api ​= 
 ​                ​if​ ​cfg!​(target_os ​=​ ​"windows"​) { 
 ​                        Backends​::​DX12 
 ​                } ​else​ ​if​ ​cfg!​(target_os ​=​ ​"macos"​) { 
 ​                        Backends​::​METAL 
 ​                } ​else​ ​if​ ​cfg!​(target_os ​=​ ​"linux"​) { 
 ​                        Backends​::​VULKAN 
 ​                } ​else​ { 
 ​                        ​panic!​(​"Unsupported platform!"​); 
 ​                }; 
  
 ​        ​// Resources 
 ​        app 
 ​                .​insert_resource​( 
 ​                        WgpuOptions { 
 ​                            backends: ​Some​(platform_api), 
  
 ​                                ..​Default​::​default​() 
 ​                        } 
 ​                );
```

**Comment by ilumar589** [comment]
@ickk @HeavyRain266 Thank you both. I've marked the first as an answer cause it was first, but thanks for the more detailed example @HeavyRain266

**Reply by heavyrain266** [reply]
Github shows that he answered 3 minutes later 🧐

**Reply by ilumar589** [reply]
For me it shows the other way around 😆

**Comment by Cannedfood** [comment]
Updated answer for bevy 0.15.0:
```rs
App::new()
    .add_plugins(DefaultPlugins.set(RenderPlugin {
        render_creation: RenderCreation::Automatic(WgpuSettings {
            #[cfg(target_os = "windows")]
            backends: Some(Backends::DX12),
            ..default()
        }),
        ..default()
    }))
```
